src/irmin-test/store.ml

```diff
@@ -102,7 +102,26 @@ module Make (S : Generic_key) = struct
         (fun () ->
           let+ _ = with_contents repo (fun t -> B.Contents.add t v2) in
           Alcotest.fail "Add after close should not be allowed")
-        (function Irmin.Closed -> Lwt.return_unit | exn -> Lwt.fail exn)
+        (function Irmin.Closed -> Lwt.return_unit 
+                (* | Index.Closed -> Lwt.return_unit 
+
+                   presumably, previously the exception thrown when operating on a closed
+                   repo was "Irmin.Closed"; after the IO refactor, the exception seems to
+                   be "Index.closed"; the library for this file store.ml (in
+                   src/irmin-test) doesn't depend on Index; thus we can't pattern match on
+                   the Index.Closed exception
+
+                   So, presumably we should catch Index.Closed everywhere and convert to
+                   Irmin.closed? 
+
+                   Catching all exceptions and returning unit seems wrong (what if the
+                   exception isn't an Index.Closed exception, but something else?)
+
+                   Another alternative is to depend on index, and catch Index.Closed (but
+                   these tests are supposed to be independent of the backend... so this
+                   can't be right).
+                *)
```

Comment:
Actually, since Irmin is here wrapping Index, presumably the repo should have a "closed" flag, and then the operation should throw an `Irmin.Closed` exception before even attempting to access the (closed) Index. So, could this simply be that we are missing the "closed flag" functionality?

Reply:
It seems to be related to https://github.com/mirage/irmin/pull/1848#discussion_r893731631 - as you say, we are not checking that the repo is closed before doing an add, however, the index does.

Reply:
we could do:
```
       B.Repo.close repo >>= fun () ->
       Lwt.catch
         (fun () ->
-          let+ _ = with_contents repo (fun t -> B.Contents.add t v2) in
-          Alcotest.fail "Add after close should not be allowed")
+          let+ _ = B.Repo.close repo in
+          Alcotest.fail "Double close should not be allowed")
         (function Irmin.Closed -> Lwt.return_unit | exn -> Lwt.fail exn)
```
that is test for double close - as for now this is only guarantee we have for when closing a repo. Or just remove this check.

Lets open an issue about forbidding operations after close and handle them later - they don't occur often.